worlds/witness/hints.py

```diff
@@ -338,105 +392,323 @@ def make_hints(world: "WitnessWorld", hint_amount: int, own_itempool: List[Item]
+    """
+    Word the hint for an area using natural sounding language.
+    This takes into account how much progression there is, how much of it is local/non-local, and whether there are
+    any local lasers to be found in this area.
+    """
 
-            already_hinted_locations.add(hint[0])
+    local_progression = sum(
+        item.player == world.player
+        and item.classification in {ItemClassification.progression, ItemClassification.progression_skip_balancing}
+        for item in corresponding_items
+    )
+
+    non_local_progression = sum(
+        item.player != world.player
+        and item.classification in {ItemClassification.progression, ItemClassification.progression_skip_balancing}
+        for item in corresponding_items
+    )
+
+    laser_names = {"Symmetry Laser", "Desert Laser", "Quarry Laser", "Shadows Laser", "Town Laser", "Monastery Laser",
+                   "Jungle Laser", "Bunker Laser", "Swamp Laser", "Treehouse Laser", "Keep Laser", }
+
+    local_lasers = sum(
+        item.player == world.player and item.name in laser_names
+        for item in corresponding_items
+    )
+
+    total_progression = non_local_progression + local_progression
+
+    player_count = world.multiworld.players
+
+    area_progression_word = "Both" if total_progression == 2 else "All"
+
+    if not total_progression:
+        hint_string = f"In the {hinted_area} area, you will find no progression items."
+
+    elif total_progression == 1:
+        hint_string = f"In the {hinted_area} area, you will find 1 progression item."
+
+        if player_count > 1:
+            if local_lasers:
+                hint_string += "\nThis item is a laser for this world."
+            elif non_local_progression:
+                other_player_str = "the other player" if player_count == 2 else "another player"
+                hint_string += f"\nThis item is for {other_player_str}."
+            else:
+                hint_string += "\nThis item is for this world."
         else:
-            hint = make_hint_from_location(world, locations_in_this_world.pop())
+            if local_lasers:
+                hint_string += "\nThis item is a laser."
+
+    else:
+        hint_string = f"In the {hinted_area} area, you will find {total_progression} progression items."
+
+        if local_lasers == total_progression:
+            sentence_end = (" for this world." if player_count > 1 else ".")
+            hint_string += f"\nAll of them are lasers" + sentence_end
+
+        elif player_count > 1:
+            if local_progression and non_local_progression:
+                if non_local_progression == 1:
+                    other_player_str = "the other player" if player_count == 2 else "another player"
+                    hint_string += f"\nOne of them is for {other_player_str}."
+                else:
+                    hint_string += f"\n{non_local_progression} of them are for other players."
```

Comment:
This one part seems to not care about player count for some reason.
```suggestion
                    other_player_str = "the other player" if player_count == 2 else "other players"
                    hint_string += f"\n{non_local_progression} of them are for {other_player_string}."
```